docs/ops/deployment/kubernetes.md

```diff
@@ -166,18 +249,34 @@ kind: Service
 metadata:
   name: flink-jobmanager
 spec:
+  type: ClusterIP
   ports:
   - name: rpc
     port: 6123
   - name: blob
     port: 6124
-  - name: query
-    port: 6125
   - name: ui
     port: 8081
   selector:
     app: flink
     component: jobmanager
 {% endhighlight %}
 
+`jobmanager-rest-service.yaml`. Optional service, which is used to add a `NodePort` on the jobmanager rest service so that user could submit job via `<public-node-ip>:<node-port>`.
```

Comment:
Can be rephrased to smt. like:
```
Optional service, that exposes the jobmanager `ui` port as public Kubernetes node's port.
```

But personally, I'd avoid adding this at all, as poor recommendation/guidance for users (as mentioned before).

Reply:
Use `NodePort` to expose an application from K8S cluster is a general solution from official doc. Why you think this should be avoided to add?

Reply:
My reasoning, that there maybe two types of readers of this docs:
 * people like devs, who have a test K8s cluster and want to setup test/experimental Flink cluster. For this, I assume they can run `kubectl port-forward|proxy` and also create resources by `kubectl`. For them the existing means should be already enough to make Flink JM available from a local dev machine. Showing them method that exposes `NodePort` would be like giving a poor example;
 * people like ops, that operate production K8s cluster. In that case they are probably knowledgeable enough how to expose a service properly.

There are some blog posts in internet, for example [this one](https://medium.com/google-cloud/kubernetes-nodeport-vs-loadbalancer-vs-ingress-when-should-i-use-what-922f010849e0) that give some information what are the different ways and in which situations to make service available outside of K8s cluster.

Reply:
Why I want to tell users how to submit jobs is inspired from this [stackoverflow question](https://stackoverflow.com/questions/56910990/flink-on-kubernetes-how-to-submit-jobs-to-session-clusters). I think it's worthy to add this instruction to offer another choice.

Reply:
I understand @1u0 concerns but I agree with @Myasuka that documenting the different ways might probably not hurt.